spec.md

```diff
@@ -290,6 +297,31 @@ encapsulated within the `data` attribute.
 * Constraints:
   * OPTIONAL
 
+### dataref
+* Type: `URI-reference`
+* Description: A reference to a location where the event payload is stored. The
+  location MAY not be accessible without further information (e.g. a pre-shared
+  secret).
+
+  Known as the "Claim Check Pattern", this attribute MAY be used for a variety
+  of purposes, including:
+
+  * If the [Data](#data) is too large to be included in the message, the `data`
+    attribute is not present, and the consumer can retrieve it using this
+    attribute.
+  * If the consumer wants to verify that the [Data](#data) has not been tampered
+    with, it can retrieve it from a trusted source using this attribute.
+  * If the [Data](#data) MUST only be viewed by trusted consumers (e.g.
+    personally identifiable information), only a trusted consumer can retrieve
+    it using this attribute and a pre-shared secret.
+
+  If this attribute is used, the information SHOULD be accessible long enough
+  for all consumers to retrieve it, but MAY not be stored for an extended period
+  of time.
+
+* Constraints:
+  * OPTIONAL
+
```

Comment:
Does the URI give information on how to obtain the data information from that URI? I assume the URI could point to different types of storing mechanism, such as block storage or DB etc.. The mechanism to retrieve the info could be different. From an event consumer's point of view, I need to know the type or how to retrieve the data info.

Reply:
For the security focused use cases, this is explicitly an anti-goal:
* Only a pre-approved consumer knows how to retrieve the data, so that no one can listen in.
* A consumer will retrieve the data from a trusted source, to make sure it hasn't been tampered with. It must explicitly not try to retrieve the data in any other way.

I agree that it would be nice for the message-too-large use case to have pre-defined mechanisms, but my guess is that it would be vendor-specific. E.g. Azure would use Blob Storage, AWS would use S3 etc. I'm not sure how to pull that off properly, or if the URI isn't already enough info to know if it is e.g. a Azure Blob or a AWS S3 reference.